packages/compiler/src/lib/decorators.ts

```diff
@@ -1022,17 +1023,8 @@ export function $withDefaultKeyVisibility(
  * ```
  */
 export function $deprecated(context: DecoratorContext, target: Type, message: string) {
-  return context.program.stateMap(deprecatedKey).set(target, message);
-}
-const deprecatedKey = createStateSymbol("deprecated");
-
-/**
- * Check if the given type is deprecated
- * @param program Program
- * @param type Type
- */
-export function isDeprecated(program: Program, type: Type): boolean {
-  return program.stateMap(deprecatedKey).has(type);
+  // TODO: Report a warning that this decorator should not be used.
+  markDeprecated(context.program, target, { message });
```

Comment:
I think we might need to keep the old deprecation stateMap as well so this is not a breaking change.

Reply:
The stateMap is still there in `deprecation.ts`, unless you mean moving its symbol to a different module would be a breaking change.

Reply:
no I guess it is fine, wrote that before I saw you kept using stateMap for it.